### PR TITLE
Track individual sources at the importer

### PR DESCRIPTION
**Pull request by tobim**
###  :notebook_with_decorative_cover: Description

Adds the following INFO messages to the node:
```
2020-12-13T23:17:05.668 importer adds zeek-reader source
2020-12-13T23:17:07.384 importer removes zeek-reader source
```

###  :memo: Checklist

- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

By commit.

### CODE REVIEW

File: libvast/src/system/importer.cpp [hunk 1]
```diff
@@ -38,6 +38,79 @@
 
 namespace vast::system {
 
+namespace {
+
+class driver : public importer_state::driver_base {
+public:
+  using super = importer_state::driver_base;
+
+  using pointer = importer_actor*;
```
Comment (tobim) [inline libvast/src/system/importer.cpp:47]: Replace by `importer_actor::pointer`.

File: libvast/src/system/importer.cpp [hunk 2]
```diff
@@ -38,6 +38,79 @@
...
+
+  pointer self() const {
+    return self_;
+  }
+
+private:
+  pointer self_;
+};
+
+class manager : public caf::detail::stream_stage_impl<driver> {
```
Comment (tobim) [inline libvast/src/system/importer.cpp:82]: s/manager/stream_stage/g

File: libvast/src/system/importer.cpp [hunk 3]
```diff
@@ -38,6 +38,79 @@
...
+public:
+  using super = caf::detail::stream_stage_impl<driver>;
+
+  manager(importer_actor* self) : caf::stream_manager(self), super(self, self) {
```
Comment (tobim) [inline libvast/src/system/importer.cpp:86]: Add a comment explaining the arguments to `super`. Don't init the `stream_manager`.

File: libvast/src/system/importer.cpp [hunk 4]
```diff
@@ -38,6 +38,79 @@
...
+    uint64_t events = 0;
+    auto t = timer::start(st.measurement_);
+    for (auto& x : xs) {
+      VAST_ASSERT(x.rows() <= static_cast<size_t>(st.available_ids()));
+      auto rows = x.rows();
+      events += rows;
+      x.offset(st.next_id(rows));
+      out.push(std::move(x));
+    }
+    t.stop(events);
```
Comment (dominiklohmann) [inline libvast/src/system/importer.cpp:67]: This PR should not change the measurement logic. The old code ran the timer per table slice, this now runs it once per vector of table slices.

It also doesn't hurt to name the input and output type here explicitly, I think, and to remove them from the `importer_state`, where they shouldn't have been in for a while now.

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

File: libvast/src/system/importer.cpp (hunk 4 above)
Comment (tobim) [reply libvast/src/system/importer.cpp:67]: I don't understand, the duration outside of the loop should be the same as the sum of the durations of the loop body?

And you mean the `using` decls? Shure that can be cleaned up.

**Review by dominiklohmann — APPROVED** [review]
I've tested this locally, and implemented the review suggestions myself that originated from a Slack call where we looked at this code together.

**Pull request merged**